src/rpc/blockchain.cpp

```diff
@@ -156,21 +156,14 @@ UniValue blockheaderToJSON(const CBlockIndex* tip, const CBlockIndex* blockindex
 
 UniValue blockToJSON(const CBlock& block, const CBlockIndex* tip, const CBlockIndex* blockindex, bool txDetails)
 {
-    // Serialize passed information without accessing chain state of the active chain!
-    AssertLockNotHeld(cs_main); // For performance reasons
```

Comment:
why is this removed?

Reply:
Because it is also part of `blockheaderToJSON`, and that is also clear immediately to readers of the code in this case.

But I was thinking about whether or not to remove it myself and was not sure.  So if you think it should be there (even though it will be asserted again in `blockheaderToJSON`), I'm happy to put it back.